git_machete/github.py

```diff
@@ -41,57 +28,129 @@ def __init__(self,
-    return GitHubPullRequest(number=int(pr_json['number']),
-                             user=pr_json['user']['login'],
-                             base=pr_json['base']['ref'],
-                             head=pr_json['head']['ref'],
-                             html_url=pr_json['html_url'],
-                             state=pr_json['state'],
-                             full_repository_name=pr_json['head']['repo']['full_name'] if pr_json['head']['repo'] else None,
-                             repository_url=pr_json['head']['repo']['html_url'] if pr_json['head']['repo'] else None)
-
-
-class GithubTokenAndTokenProvider(NamedTuple):
-    token: str
-    token_provider: str
+class RemoteAndOrganizationAndRepository:
+    def __init__(self,
+                 remote: Optional[str],
+                 organization: Optional[str],
+                 repository: Optional[str]
+                 ) -> None:
+        self.remote = remote
+        self.organization = organization
+        self.repository = repository
+
+    def __bool__(self) -> bool:
+        return self.remote is not None and self.organization is not None and self.repository is not None
+
+    @classmethod
+    def from_config(cls,
+                    git: GitContext
+                    ) -> "RemoteAndOrganizationAndRepository":
+        return RemoteAndOrganizationAndRepository(remote=git.get_config_attr_or_none(key=git_config_keys.GITHUB_REMOTE),
+                                                  organization=git.get_config_attr_or_none(key=git_config_keys.GITHUB_ORGANIZATION),
+                                                  repository=git.get_config_attr_or_none(key=git_config_keys.GITHUB_REPOSITORY))
+
+    @classmethod
+    def from_url(cls,
+                 domain: str,
+                 url: str,
+                 remote: str
+                 ) -> Optional["RemoteAndOrganizationAndRepository"]:
+        for pattern in github_remote_url_patterns(domain):
+            match = re.match(pattern, url)
+            if match:
+                org = match.group(1)
+                repo = match.group(2)
+                return RemoteAndOrganizationAndRepository(remote=remote,
+                                                          organization=org,
+                                                          repository=repo if repo[-4:] != '.git' else repo[:-4])
+        return RemoteAndOrganizationAndRepository(remote=None,
+                                                  organization=None,
+                                                  repository=None)
+
+
+class GitHubToken:
+    GITHUB_TOKEN_ENV_VAR = 'GITHUB_TOKEN'
 
+    def __init__(self,
+                 domain: str
+                 ) -> None:
+        self.__domain = domain
+        self.__value: Optional[str] = None
+        self.__provider: Optional[str] = None
+        for token_retrieval_method in [self.__get_token_from_hub, self.__get_token_from_gh, self.__get_token_from_env,
```

Comment:
The order has changed ‼️ I'll fix that now, but generally https://github.com/VirtusLab/git-machete/issues/763 FTW